services/app-api/utils/auth/authorization.ts

```diff
@@ -51,17 +51,17 @@ export const isAuthenticated = async (event: APIGatewayProxyEvent) => {
     clientId: cognitoValues.userPoolClientId,
   });
 
-  let isAuthenticated;
+  let isAuthenticated = false;
   if (event?.headers?.["x-api-key"]) {
     try {
-      isAuthenticated = await verifier.verify(event.headers["x-api-key"]);
+      await verifier.verify(event.headers["x-api-key"]);
+      isAuthenticated = true;
     } catch {
-      // verification failed - unauthenticated
-      isAuthenticated = false;
+      // verification failed - ignore and let it return false
     }
   }
 
-  return !!isAuthenticated;
+  return isAuthenticated;
```

Comment:
Suggestion, but non-blocking: My thought here is to go even further, if you've got the appetite. My priorities to improve maintainability are always to reduce (in order of importance): state, coupling, complexity, code. We could remove a tiny bit of state here by eliminating the `isAuthenticated` variable:

```ts
  try {
    await verifier.verify(event?.headers?.["x-api-key"]!);
    return true;
  } catch {
    return false;
  }
```

I am relying on `verifier.verify(undefined)` and `verifier.verify("")` to error out here, which I have not tested. Otherwise we may still need a guard clause up front:
```ts
  if (!event?.headers?.["x-api-key"]) {
    return false;
  }
```

Reply:
Good stuff! I tested with `undefined` and with empty string. Both triggered the catch block.

Reply:
I'll also note, for our own reassurance, this matches the [example the aws-jwt-verify package gives](https://www.npmjs.com/package/aws-jwt-verify), where any invalid token will throw an error. If it does not throw, it is considered to have passed validation.